File: CodeRun/medium/pallets.py
n = int(input())
pallets = []
for i in range(n):
    a, b = tuple(map(int, input().split()))
    if a > b:
        pallets.append((a, b))
    else:
        pallets.append((b, a))
pallets.sort(reverse=True)
res = 1
OldBig = pallets[0]
Big = pallets[0]
for i in range(1, n):
    if not (pallets[i][0] < Big[0] and pallets[i][1] < Big[1] or pallets[i][0] < OldBig[0] and pallets[i][1] < OldBig[1]):
        res += 1
        if pallets[i][1] > Big[1]:
            OldBig = Big
            Big = pallets[i]
print(res)

File: CodeRun/medium/pallets.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int n;
    std::cin >> n;
    std::vector<std::pair<long, std::pair<long, long>>> pallets(n);
    for (int i = 0; i < n; i++) {
        int a, b;
        std::cin >> a;
        std::cin >> b;
        pallets[i] = std::make_pair(a + b, std::make_pair(a, b));
    }
    std::sort(pallets.begin(), pallets.end());
    long res = 0;
    for (int i = 0; i < n; i++) {
        bool can = false;
        for (int j = i + 1; j < n; j++) {
            long f_i = pallets[i].second.first;
            long s_i = pallets[i].second.second;
            long f_j = pallets[j].second.first;
            long s_j = pallets[j].second.second;
            if ((f_i < f_j && s_i < s_j) || (f_i < s_j && s_i < f_j)) {
                can = true;
                break;
            }
        }
        if (!can) {
            res += 1;
        }
    }
    std::cout << res << std::endl;
    return 0;
}